fastlane_core/lib/fastlane_core/certificate.rb

```diff
@@ -0,0 +1,37 @@
+
+module FastlaneCore
+  class Certificate
+    class << self
+      # @return (Hash) The hash with the data of the certificate
+      # @example
+      #  {"SerialNumber"=>"069FBA785158DC46D9A6603EAF5A1316"
+      #  "FileName"=>"certificate-13",
+      #  "NotAfter"=>"Feb 21 16:39:53 2024 GMT",
+      #  "NotBefore"=>"Feb 21 16:39:53 2023 GMT"}
+      def order_by_expiration(certs, ascending = true)
+        certs.sort_by { |c| c["NotAfter"].to_i }.reverse! unless ascending
+        certs.sort_by { |c| c["NotAfter"].to_i }
+      end
+
+      def parse_from_b64(b64_content)
+        openssl_object = OpenSSL::X509::Certificate.new(Base64.decode64(b64_content))
+        {
+          "SerialNumber" => openssl_object.serial.to_s(16).upcase,
+          "NotBefore" => openssl_object.not_before,
+          "NotAfter" => openssl_object.not_after
+        }
+      end
+
+      def parse_from_file(path)
+        file_name = File.basename(path).gsub(".cer", "").gsub(".p12", "").gsub(".pem", "")
```

Comment:
Not sure why but originally in runner.rb (commit 8b2cd4f) this was done and the filename without the extension was passed along to generate_profile etc. so i just kept it like that. I did not name it certificate_id (like it was named before) because its a bit misleading. I could maybe keep the file extension here and remove it just before passing it on in runner.rb. Would that be better?

Reply:
Yea, it's hard to tell why that was done. I think it would be helpful to understand if it is necessary or not. Looking at the code I just see it used in one place to output a log, but it's possible it's coming back from an Apple API that way, so it could be done to align with that?

Reply:
After some digging i found that its used to check if a cert thats on apple dev portal is available locally. Certs created by Spaceship have the cert_id as their file names (its the only way to actually get the cert_id and was used to identify the certs). Its not even being used anymore since the serial number is the "main" identifier now so i think it can be removed to avoid confusion.

Reply:
I think it can be safely removed since the log message doesnt say much anyways and the serial number is the main way to identify certs now.

Reply:
Actually nope, it is available as an input parameter to the sigh tool. I dont understand how it would be useful in practice though. The one and only way to get the cert id is when generating the certificate through the apple api (spaceship) which doesnt seem very practical. I guess we can just leave it as it is now.

Reply:
Okay, thanks for investigating. Would you mind adding a comment in the code with some details on where it's used, etc.? That way we have at least a little context there if we want to come back and refactor it in the future.